### DISCUSSION TITLE
WooCommerce Analytics: Custom Order Statuses

### DISCUSSION THREAD

**Discussion by masteradhoc** [woocommerce/woocommerce#52638] [General]
Is there an easy way in the Analytics area (/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fsettings) to make any custom statuses automatically selectable via hook/filter?

We have a plugin that added a custom status but was not recognized in the analytics area. In order to fix this automatically for the customers, we always wanted to select this automatically.

Thanks for your feedback

![image](https://github.com/user-attachments/assets/5f799441-400c-4825-9559-6ebfcd090552)

**Comment by joshuatf** [comment]
Hey @masteradhoc 👋 

The statuses in the screenshot you provided are pulled from `wc_get_order_statuses`.  If you're using the `wc_order_statuses` filter to add your custom filter, they should appear in that list:

```php
function add_custom_status( $order_statuses ) {
	$order_statuses['wc-test-status'] = 'Test Status';
	return $order_statuses;
}
add_filter( 'wc_order_statuses', 'add_custom_status' );
```

> to make any custom statuses automatically selectable via hook/filter?

To clarify the issue here, are you trying to add a status so that it can be selected or are you trying to force it on by default?  For the latter, there's not a dedicated way to do this yet but we could either:

1. Open an issue to introduce a filter so that default statuses can be enabled.
2. You could work around this right now by manually updating the `woocommerce_actionable_order_statuses` option in the `wp_options` table to include your status.

Let me know if I misunderstood your question and if you need any further help with this.

**Reply by masteradhoc** [reply]
@joshuatf thanks for the fast answer here!

_To clarify the issue here, are you trying to add a status so that it can be selected or are you trying to force it on by default For the latter, there's not a dedicated way to do this yet but we could either:_

We try to force our custom status to be on by default for analytics. As this doest seem possible then at the moment i'll open a ticket as suggested by you. Keep this thread posted.

**Comment by masteradhoc** [comment]
New Issue opened at #53323. Thanks again @joshuatf - hope we could get this tweak anytime soon :)
